app/models/furniture.rb

```diff
@@ -24,6 +24,20 @@ class Furniture < ApplicationRecord
 
   validates :furniture_kind, presence: true
 
+  # Used by child classes that require storage of secrets, like Square API keys
+  # for a Marketplace
+  has_encrypted :secrets, type: :json
+
+  # Setting `secrets` to an empty hash after initialization will force consistent access for developers, rather than having to code around a potential nil value
+  after_initialize do
+    # The `RankedModel` gem uses
+    #   [`ActiveRecord::QueryMethods.select`](https://api.rubyonrails.org/classes/ActiveRecord/QueryMethods.html#method-i-select)
+    # to do some stuff (Validation?) under the covers while keeping memory usage light... Which is good!
+    # But it also means that we don't have the ciphertext!
+    # So we can't set the secrets in that case (nor would we want to)
+    self.secrets ||= {} if has_attribute?("secrets_ciphertext")
```

Comment:
@zspencer Makes me think this will not be the last time we try to access data on models returned ranked queries with surprising results 🤣